python_modules/dagster-test/dagster_test/fixtures/docker_compose.py

```diff
@@ -192,14 +194,24 @@ def buildkite_hostnames_cm(network):
         disconnect_container_from_network(container, network)
 
 
-def default_docker_compose_yml(default_directory):
+def default_docker_compose_yml(default_directory) -> str:
     if os.path.isfile("docker-compose.yml"):
         return os.path.join(os.getcwd(), "docker-compose.yml")
     else:
         return os.path.join(default_directory, "docker-compose.yml")
 
 
-def network_name_from_yml(docker_compose_yml):
-    dirname = os.path.dirname(docker_compose_yml)
-    basename = os.path.basename(dirname)
-    return basename + "_default"
+def network_name_from_yml(docker_compose_yml) -> str:
+    with open(docker_compose_yml) as f:
+        config = yaml.safe_load(f)
+    if "name" in config:
+        name = config["name"]
+    else:
+        dirname = os.path.dirname(docker_compose_yml)
+        name = os.path.basename(dirname)
+    if "networks" in config:
+        network_name = next(iter(config["networks"].keys()))
```

Comment:
Do you foresee anything unintended happening if there are multiple and we just grab the first?

Reply:
so i think you have more than one network you're already doing something kind of complicated that I dont want to predict, so the downside of not hardfailing is that maybe you're not and want to know sooner that this wont' work, the benefit is that in that case you don't need to come make a change here to convince this logic tha tyou're doing something intended. in my estimate it's probably more likely that if you have multiple networks you're doing somethign that i can't really predict here and so this is fine?